labeling/codebase/localcodebase.go

```diff
@@ -14,23 +15,39 @@ type LocalCodebase struct {
 func (c LocalCodebase) FindFileMatching(
 	predicate func(string) bool,
 	glob ...string,
-) (string, error) {
-	for _, g := range glob {
-		filesFound, err := filepath.Glob(filepath.Join(c.BasePath, g))
-
-		if err != nil {
-			continue
-		}
-
-		for _, path := range filesFound {
-			relPath, err := filepath.Rel(c.BasePath, path)
-			if predicate(relPath) {
-				return relPath, err
+) (foundPath string, err error) {
+	basePath := c.BasePath
+	if basePath == "" {
+		basePath = "."
+	}
+	err = filepath.WalkDir(
+		basePath,
+		func(path string, d fs.DirEntry, fileError error) error {
+			if fileError != nil {
+				return fileError
+			}
+			for _, g := range glob {
+				relPath, innerErr := filepath.Rel(c.BasePath, path)
```

Comment:
I'm not clear on how we'd get to an unreachable path via walkdir. Also, should this check be done once per path, instead of for each glob (since it's not dependent on the glob)?

Reply:
`filepath.Rel` only returns an error if:
>  targpath can't be made relative to basepath or if knowing the current working directory would be necessary to compute it.

That should never be the case because WalkDir will only walk dirs that can be relative to the targpath, and only emit absolute dirs.

Changed it to return the error instead, doesn't harm.

Also good catch on moving it out of the loop.